### ISSUE TITLE
nomacs could not change theme through Setting Tab

### ISSUE THREAD

**Issue by LloydLore** [nomacs/nomacs#1062]
I am a new user for nomacs v3.172295. The default theme is "Dark-theme", and I'd like to use "Light-theme". 
Issue Description: Could not change theme through menu/settings tab, 
Reproduce Step: 
1. Edit --> Settings, (Ctrl+Shift+P, could enter setting tab as well)
2. In settings tab, select General tab, and choose the option "Color Settings" from "Dark Theme" to "Light Theme"
3. The bottom line of this tab will prompt "Please Restart nomacs to apply changes"
4. Click the prompt, the application restarts, and the theme doesn't change
5. Also, the option "Color Setting"'s value doesn't change.

Walkaround:
I could manually change the settings resides in path "$HOME/.config/nomacs/'Image Louge.conf', the config field is "themeName312", change this value will result in application theme change. 

BTW, this application is awesome, thank you!

**Comment by Aqa-Ib** [comment]
Can reproduce, but in my case the default theme is light and I can't change it to neither dark nor system.
Thanks!

**Comment by leejuyuu** [comment]
It seems that this bug is intermittent. Some times the theme and config file is written correctly.

**Comment by leejuyuu** [comment]
@LloydLore @Aqa-Ib I had a harder time reproducing this on 3.17.2296 (515812d) when trying to fix this. Could you try newer version as well?

A possible cause is that `QSettings` does not write to the file before the program closes. Maybe we can call `sync()` manually to try to make file writes earlier.

**Comment by LloydLore** [comment]
@leejuyuu , I am using Arco Linux while it seems the package manager didn't update to the newest version. I will try it once the package is updated.

**Comment by Aqa-Ib** [comment]
I have tried with latest git (installed through AUR package nomacs-git) and still the same here.
Probably these warnings are relevant:
```
[WARNING] QMetaObject::connectSlotsByName: No matching signal for on_themeBox_currentIndexChanged(QString)
[WARNING] QMetaObject::connectSlotsByName: No matching signal for on_keepZoom_buttonClicked(int)
[WARNING] QMetaObject::connectSlotsByName: No matching signal for on_loadGroup_buttonClicked(int)
[WARNING] QMetaObject::connectSlotsByName: No matching signal for on_saveGroup_buttonClicked(int)
[WARNING] QMetaObject::connectSlotsByName: No matching signal for on_openDefault_clicked()
[WARNING] QMetaObject::connectSlotsByName: No matching signal for on_associateFiles_clicked()
[WARNING] QMetaObject::connectSlotsByName: No matching signal for on_loadRaw_buttonClicked(int)
```

**Comment by leejuyuu** [comment]
> I have tried with latest git (installed through AUR package nomacs-git) and still the same here.

@Aqa-Ib Thanks for providing the watning messages.
Please help provide the version of nomacs you can find in “？ > About”.
It would be hard to identify "the latest git version" for future issue readers because the project is still being developed.

**Comment by Aqa-Ib** [comment]
My build is from commit e9a25e560e218713d1fdf1876aee20b858adb0a7

**Comment by artofit** [comment]
I have the same issue( does not save settings) with a fresh install of Manjaro XFCE 24.0
Nomacs  3.17.2295(Qt 6.7.0)

**Comment by LloydLore** [comment]
> I have the same issue( does not save settings) with a fresh install of Manjaro XFCE 24.0 Nomacs 3.17.2295(Qt 6.7.0 ; [94e9ada](https://github.com/nomacs/nomacs/commit/94e9adaaeb089686809730bbff058cebf7f7e9c3) 2024-02-29)
> 
> The build Nomacs 3.17.2295 with **Qt 5.15.13 ([6dfc779](https://github.com/nomacs/nomacs/commit/6dfc77904b1aa4b2caa8a0d18f10bd9e5406de52), 2024-05-20) solves the issues !!** So probably a Qt6 bug
> 
> REM: under "/.config/nomacs/'Image Louge.conf" I do not see/have a theme field as said in first post

The theme field lies under section `[DisplaySettings]` and the field is `themeName312`

**Comment by artofit** [comment]
> 
> The theme field lies under section `[DisplaySettings]` and the field is `themeName312`
With Qt 5.15.13 I have indeed under "/.config/nomacs/'Image Louge.conf" :
**themeName312=Dark-Theme.css**
@LloydLore do you mean if I revert back to Qt 6.7.x and manually modify as it'll be ok?

**Comment by LloydLore** [comment]
@artofit , on my machine, nomacs is build on top of Qt  6.6.3, 
see below:
![image](https://github.com/nomacs/nomacs/assets/17218122/772a0147-758e-43b1-9fdd-1dde2e32c531)
and I manually set the theme as such:
![image](https://github.com/nomacs/nomacs/assets/17218122/e9302ae2-9222-4789-a616-3bf093e1d421)
I shall say it works on my machine, and I don't tried this method with Qt 6.7.x.

**Comment by leejuyuu** [comment]
Thanks for the input! Confirmed it's related to the Qt version and the missing signals.

**Comment by leejuyuu** [comment]
@novomesk I think these missing signals appears because some things got renamed at some point. I suggest removing the autoconnecting `QMetaObject::connectSlotsByName` and replace with manually writing `QObject::connect`, so we can (probably) catch some of the renaming problems at compile time. Do you think this is a valid approach?

**Comment by novomesk** [comment]
> Do you think this is a valid approach?

Honestly, I don't know. I am not familiar with this part of code.

**Comment by leejuyuu** [comment]
Thanks for the response! I'll try what I can do then.

**Comment by leejuyuu** [comment]
@LloydLore, @Aqa-Ib, @artofit  If you are available, please help test whether #1096 fixes this issue (and if it breaks other things). Thanks!
